be/src/olap/olap_cond.cpp

```diff
@@ -651,14 +651,10 @@ bool Conditions::delete_conditions_eval(const RowCursor& row) const {
 bool Conditions::rowset_pruning_filter(const std::vector<KeyRange>& zone_maps) const {
     //通过所有列上的删除条件对version进行过滤
     for (auto& cond_it : _columns) {
-        if (_cond_column_is_key_or_duplicate(cond_it.second) && cond_it.first > zone_maps.size()) {
-            LOG(WARNING) << "where condition not equal zone maps size. "
-                         << "cond_id=" << cond_it.first
-                         << ", zone_map_size=" << zone_maps.size();
-            return false;
-        }
-        if (_cond_column_is_key_or_duplicate(cond_it.second) && !cond_it.second->eval(zone_maps.at(cond_it.first))) {
-            return true;
+        if (_cond_column_is_key_or_duplicate(cond_it.second)) {
+            if (cond_it.first < zone_maps.size() && !cond_it.second->eval(zone_maps.at(cond_it.first))) {
```

Comment:
it cores just because of `zone_maps.size() == cond_it.first`  and `zone_maps.at(cond_it.first)` returns null ?

Reply:
Yes, because zone_maps has only one item. zone_maps.at(1) will core dump.